src/server/scripts/EasternKingdoms/TheStockade/boss_lord_overheat.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2008-2017 TrinityCore <http://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "the_stockade.h"
+
+enum Spells
+{
+    SPELL_FIREBALL = 12466, //starts 1-2 secs from pull
+    SPELL_OVERHEAT = 86633, //probably cast every 10 secs, need to confirm.
+    SPELL_RAIN_OF_FIRE = 86636  //probably cast every 10 secs, need to confirm
+};
+
+enum Events
+{
+    EVENT_OVERHEAT = 1,
+    EVENT_RAIN_OF_FIRE,
+    EVENT_FIREBALL
+};
+
+enum Says
+{
+    SAY_PULL, //Yell: ALL MUST BURN!
+    SAY_DEATH, //Yell: FIRE... EXTINGUISHED!
+
+};
+
+class boss_lord_overheat : public CreatureScript
+{
+public:
+    boss_lord_overheat() : CreatureScript("boss_lord_overheat") {}
+
+    struct boss_lord_overheatAI : public BossAI
+    {
+        boss_lord_overheatAI(Creature* creature) : BossAI(creature, DATA_LORD_OVERHEAT) {}
+
+        void Reset() override
+        {
+            _Reset();
+        }
+        void EnterCombat(Unit* /*who*/) override
+        {
+            _EnterCombat();
+            Talk(SAY_PULL);
+            events.ScheduleEvent(EVENT_FIREBALL, Seconds(2));
+            events.ScheduleEvent(EVENT_OVERHEAT, randtime(Seconds(9), Seconds(11)));
```

Comment:
you dont need use this randtime()

events.ScheduleEvent(EVENT_OVERHEAT, Seconds(9), Seconds(11));